### PR TITLE
Virtual Kubelet Node controller 1/6

### PR DESCRIPTION
**Pull request by mlavacca**
# Description

This PR introduces the following changes:
1. `logrus` has been entirely replaced by `klog` in all the virtual kubelet code
2. the `trace` package has been removed
3. the node controller has been moved from `internal` to `pkg`

address #474

### CODE REVIEW

File: pkg/virtualKubelet/node/node.go [hunk 1]
```diff
@@ -216,7 +214,7 @@ func (n *NodeController) Run(ctx context.Context) error {
 	}
 	n.lease = l
 
-	log.G(ctx).Debug("Created node lease")
+	klog.V(4).Info("Created node lease")
```
Comment (palexster) [inline pkg/virtualKubelet/node/node.go:217]: I would add a message to specify which node is concerned by this log

File: pkg/virtualKubelet/node/node.go [hunk 2]
```diff
@@ -274,7 +272,7 @@ func (n *NodeController) controlLoop(ctx context.Context) error {
 				t = statusTimer
 			}
 
-			log.G(ctx).Debug("Received node status update")
+			klog.V(4).Info("Received node status update")
```
Comment (palexster) [inline pkg/virtualKubelet/node/node.go:275]: ditto

File: pkg/virtualKubelet/node/node.go [hunk 3]
```diff
@@ -385,31 +377,18 @@ func ensureLease(ctx context.Context, leases v1beta1.LeaseInterface, lease *coor
...
-
-	if lease.Spec.LeaseDurationSeconds != nil {
-		ctx = span.WithField(ctx, "lease.expiresSeconds", *lease.Spec.LeaseDurationSeconds)
-	}
-
 	l, err := leases.Update(context.TODO(), lease, metav1.UpdateOptions{})
 	if err != nil {
 		if errors.IsNotFound(err) {
-			log.G(ctx).Debug("lease not found")
+			klog.V(4).Info("lease not found")
```
Comment (palexster) [inline pkg/virtualKubelet/node/node.go:383]: ditto

File: pkg/virtualKubelet/node/node.go [hunk 4]
```diff
@@ -385,31 +377,18 @@ func ensureLease(ctx context.Context, leases v1beta1.LeaseInterface, lease *coor
...
 			l, err = ensureLease(ctx, leases, lease)
 		}
 		if err != nil {
-			span.SetStatus(err)
 			return nil, err
 		}
-		log.G(ctx).Debug("created new lease")
+		klog.V(4).Info("created new lease")
```
Comment (palexster) [inline pkg/virtualKubelet/node/node.go:389]: ditto

File: pkg/virtualKubelet/node/node.go [hunk 5]
```diff
@@ -385,31 +377,18 @@ func ensureLease(ctx context.Context, leases v1beta1.LeaseInterface, lease *coor
...
 	} else {
-		log.G(ctx).Debug("updated lease")
+		klog.V(4).Info("updated lease")
```
Comment (palexster) [inline pkg/virtualKubelet/node/node.go:391]: ditto

File: pkg/virtualKubelet/node/node.go [hunk 6]
```diff
@@ -463,35 +442,25 @@ func preparePatchBytesforNodeStatus(nodeName types.NodeName, oldNode *corev1.Nod
...
-
 	var node *corev1.Node
 
 	oldNode, err := nodes.Get(context.TODO(), n.Name, emptyGetOptions)
 	if err != nil {
 		return nil, err
 	}
 
-	log.G(ctx).Debug("got node from api server")
+	klog.V(4).Info("got node from api server")
```
Comment (palexster) [inline pkg/virtualKubelet/node/node.go:452]: ditto

File: pkg/virtualKubelet/node/sync.go [hunk 7]
```diff
@@ -54,15 +52,15 @@ func (p *syncProviderWrapper) _deletePodKey(ctx context.Context, key string) {
...
+	klog.V(4).Info("syncProviderWrappper.DeletePod")
 	key, err := cache.MetaNamespaceKeyFunc(pod)
 	if err != nil {
 		return err
 	}
 
 	p.deletedPods.Store(key, pod)
 	if err := p.PodLifecycleHandler.DeletePod(ctx, pod.DeepCopy()); err != nil {
-		log.G(ctx).WithField("key", key).WithError(err).Debug("Removed key from deleted pods cache")
+		klog.V(4).Info("%v - Removed key %s from deleted pods cache", err, key)
```
Comment (palexster) [inline pkg/virtualKubelet/node/sync.go:63]: ditto

**Review by palexster — APPROVED** [review]

**Review by aleoli — APPROVED** [review]

**Pull request merged**